core/src/test/java/uk/gov/gchq/koryphe/impl/function/IterableFunctionTest.java

```diff
@@ -92,4 +94,33 @@ public void shouldReturnEmptyIterableForEmptyInput() {
         assertNotNull(result);
         assertTrue(Iterables.isEmpty(result));
     }
+
+    @Test
+    public void shouldPopulateFunctionFromBuilder() {
+        // Given
+        final IterableFunction<Integer, String> function = new IterableFunction.Builder<Integer>()
+                .first(Object::toString)
+                .then(Integer::valueOf)
+                .then(Object::toString)
+                .build();
+
+        // Then
+        assertEquals(3, function.getFunctions().size());
```

Comment:
It would be good to test the actual functions values are as expected. For example, create variables for the functions (Object::toString, etc.) then:
```java
assertEquals(Arrays.asList(func1, func2, func3), function.getFunctions9));
```